feat: support Pydanticv2

spumer opened the pull request:
BREAKING CHANGE: DROP PYDANTIC V1 SUPPORT

Fixes #50

spumer commented:
@smagafurov i think we can publish 3.0.0b from this branch. This allow others start testing their setups

smagafurov commented on tests/test_openapi.py:
```diff
@@ -661,18 +658,33 @@ def app():
+            'api__mobile___Params_probe_',
+            'api__mobile___Request_probe_',
+            'api__web___Params_probe_',
+            'api__web___Request_probe_',
```
may be we can repeat naming style as 'api__web___Request[probe]' ?

smagafurov commented on tests/test_openapi_dependencies.py:
```diff
@@ -605,68 +585,7 @@ def test_basic(app_client, openapi_compatible):
-                            'description': '[-32603] Internal error\n\nInternal JSON-RPC error',
-                        },
-                    },
+                    'responses': ANY,
```
не осилил? :)

spumer replied:
Да я устал копипастить одно и то же, оно для этого теста не значимо, тк тут проверяется правильно ли работает Dependencies, а responses не важны. Так что да ):

spumer commented on tests/test_jsonrpc.py:
```diff
@@ -125,8 +132,8 @@ def test_request_id_validation_error(echo, json_request):
-                {'loc': ['id'], 'msg': 'str type expected', 'type': 'type_error.str'},
-                {'loc': ['id'], 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+                {'input': [123], 'loc': ['id', 'str'], 'msg': 'Input should be a valid string', 'type': 'string_type'},
+                {'input': [123], 'loc': ['id', 'int'], 'msg': 'Input should be a valid integer', 'type': 'int_type'}
```
it's new format for errors

loc now mean field name and concrete annotation:

for declaration `id: Union[StrictStr, int, float, decimal.Decimal]` it will be

```
[
{'type': 'string_type', 'loc': ('id', 'str'), 'msg': 'Input should be a valid string', 'input': [123]},
{'type': 'int_type', 'loc': ('id', 'int'), 'msg': 'Input should be a valid integer', 'input': [123]},
{'type': 'float_type', 'loc': ('id', 'float'), 'msg': 'Input should be a valid number', 'input': [123]},
{'type': 'decimal_type', 'loc': ('id', 'decimal'), 'msg': 'Decimal input should be an integer, float, string or Decimal object', 'input': [123]}
]
```

spumer replied:
i resolve it, now loc pointing to field location in json and type (annotation) info cutted

sam-courtemanche-wmx commented:
Hey, nice work on this, so far my tests on this version have had really nice results.

I have noticed one small issue though:

When there is no Entrypoint in an app, the documentation cannot be generated anymore.
`/docs`, `/redos` and `/openapi.json` all raise the same error

```python3
  File ".../site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File ".../site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File ".../site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File ".../site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File ".../site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File ".../site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File ".../site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File ".../site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File ".../site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File ".../site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File ".../site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/fastapi/applications.py", line 247, in openapi
    return JSONResponse(self.openapi())
                        ^^^^^^^^^^^^^^
  File ".../site-packages/fastapi_jsonrpc/__init__.py", line 1336, in openapi
    for key, schema in result['components']['schemas'].items():
                       ~~~~~~^^^^^^^^^^^^^^
KeyError: 'components'
```

spumer commented:
Oups! Thanks for report!

spumer commented:
Rebased from master, fixed issue from https://github.com/smagafurov/fastapi-jsonrpc/pull/53#issuecomment-1742549617,  Support pydantic v2.4

@smagafurov please publish new version - 3.0.0b2

smagafurov commented:
> @smagafurov please publish new version - 3.0.0b2

done

spumer commented:
I think we can wait https://github.com/smagafurov/fastapi-jsonrpc/pull/56

Cause openRPC support looks like good feature to have it in both versions
